google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAdCustomizer.java

```diff
@@ -0,0 +1,498 @@
+  /** We're doing only searches by resource_name in this example, we can set page size = 1. */
+  private static final int PAGE_SIZE = 1;
+
+  private static class AddAdCustomizerParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private List<Long> adGroupIds;
+
+    @Parameter(names = ArgumentNames.FEED_NAME, required = true)
+    private String feedName;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddAdCustomizerParams params = new AddAdCustomizerParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupIds = Arrays.asList(
+          Long.parseLong("INSERT_AD_GROUP_ID_HERE"),
+          Long.parseLong("INSERT_AD_GROUP_ID_HERE"));
+      params.feedName = "INSERT_FEED_NAME_HERE";
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddAdCustomizer().runExample(
+          googleAdsClient, params.customerId, params.adGroupIds, params.feedName);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupIds the ad group IDs.
+   * @param feedName the name of the feed to create.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, List<Long> adGroupIds, String feedName) {
+
+    // Makes sure the feed name is not duplicated.
+    feedName = feedName + System.currentTimeMillis();
```

Comment:
Can't we just make this something random to begin with, rather than taking a CLI flag? Like call it `"Example ad customizers " + System.currentTimeMillis()`.

Reply:
Right, makes sense. also removed the `feedName` argument as it was no longer needed.